conda/common/logic.py

```diff
@@ -73,17 +246,21 @@ def from_index(self, m):
         return self.indices.get(m)
 
     def Assign_(self, vals, name=None):
-        tvals = type(vals)
-        if tvals is tuple:
-            x = self.new_var()
-            self.clauses.extend((-x,) + y for y in vals[0])
-            self.clauses.extend((x,) + y for y in vals[1])
```

Comment:
These `tuple` additions and generator expressions are surprisingly costly!
To entirely avoid them, I did all some ugly
```python
if add_new_clauses:
 ...
```
code duplications below. Those are really ugly, but removing them will increase the total runtime of `LinearBound_` by approx. `60 %`, IIRC.